packages/build-ui/src/pages/collection/collection_name_modal.rs

```diff
@@ -0,0 +1,113 @@
+use bdk::prelude::*;
+
+#[component]
+pub fn CollectionNameModal(
+    show: bool,
+    on_back: EventHandler<()>,
+    on_add: EventHandler<String> // Changed to pass the collection name
+) -> Element {
+    let mut collection_name = use_signal(|| String::new());
+    
+    if !show {
+        return rsx!(div {});
+    }
+
+    rsx! {
+        // Modal backdrop with purple glow effect
+        div { 
+            class: "fixed inset-0 bg-black bg-opacity-50 backdrop-blur-sm z-50",
+            style: "background: radial-gradient(circle at center, rgba(76, 29, 149, 0.15) 0%, rgba(0, 0, 0, 0.5) 100%)",
+            onclick: move |_| on_back.call(()),
```

Comment:
Refer to a way to utilize our popup service on Ratel.
- Define popup layout: https://github.com/biyard/ratel/blob/dev/packages/main-ui/src/pages/layout.rs#L81
- Open popup: https://github.com/biyard/ratel/blob/dev/packages/main-ui/src/pages/components/header.rs#L131